app/src/main/java/org/thoughtcrime/securesms/preferences/appearance/AppearanceSettingsViewModel.kt

```diff
@@ -26,11 +27,17 @@ class AppearanceSettingsViewModel @Inject constructor(private val prefs: TextSec
         prefs.setThemeStyle(newThemeStyle)
         // update UI state
         _uiState.value = prefs.themeState()
+
+        // force compose to refresh its style reference
+        selectedTheme = null
     }
 
     fun setNewFollowSystemSettings(followSystemSettings: Boolean) {
         prefs.setFollowSystemSettings(followSystemSettings)
         _uiState.value = prefs.themeState()
+
+        // force compose to refresh its style reference
+        selectedTheme = null
```

Comment:
This won't force currently composed layouts to refresh, it'll just refresh on a subsequent call.

So this Activity (and all open activities in the back stack) won't update.

Maybe [this PR](https://github.com/bemusementpark/session-android/pull/5/files#diff-7c7c8d6c3e67d4fd368263cff0674e967fd96366a8fcf928f5d59c16cc40a6d5R393) will help.

Reply:
In a fully compose setup the value will come from a stateflow, directly passed to the root composable.
For now it works, even with the back stack, as going back to a previous fragment/activity calls the composable again which will check the value.
I'm a bit wary of moving all our composable theming in a module where it doesn't belong...

Reply:
oh yeah... `EmptyView` is recomposed when you return to `HomeActivity`, I don't know why, I don't know if it's reliable.

Yeah, we don't have to move compose components. We can keep compose in this module, We'd just probably need to make another object... or just `@Provides` this `StateFlow<Theme>` with Dagger.

Reply:
The currently open activity won't update though. It doesn't have any compose now...

Reply:
Once we move to a fully composed screen the theme will be passed via a Flow so this won't be an issue anymore.
If we change this screen first before changing the full app we can make that flow available and collected at the root composable here to be applied.